extensions/markdown/src/documentLinkProvider.ts

```diff
@@ -44,26 +41,22 @@ export default class MarkdownDocumentLinkProvider implements vscode.DocumentLink
 		return results;
 	}
 
-	private normalizeLink(link: string, base: string, toc: TableOfContentProvider): vscode.Uri {
-		let uri = vscode.Uri.parse(link);
-		if (!uri.scheme) {
-			if (uri.fragment && !uri.path) {
-				// local link
-				const line = toc.lookup(uri.fragment);
-				if (!isNaN(line)) {
-					return vscode.Uri.parse(`command:revealLine?${encodeURIComponent(JSON.stringify({ lineNumber: line, at: 'top' }))}`);
-				}
-			}
+	private normalizeLink(document: vscode.TextDocument, link: string, base: string): vscode.Uri {
+		const uri = vscode.Uri.parse(link);
+		if (uri.scheme) {
+			return uri;
+		}
 
-			// assume it must be a file
-			let file;
-			if (uri.path[0] === '/') {
-				file = path.join(vscode.workspace.rootPath, uri.path);
-			} else {
-				file = path.join(base, uri.path);
-			}
-			uri = vscode.Uri.file(file);
+		// assume it must be a file
+		let resourcePath;
+		if (!uri.path) {
+			resourcePath = document.uri.path;
+		} else if (uri.path[0] === '/') {
+			resourcePath = path.join(vscode.workspace.rootPath, uri.path);
+		} else {
+			resourcePath = path.join(base, uri.path);
 		}
-		return uri;
+
+		return vscode.Uri.parse(`command:_markdown.openDocumentLink?${encodeURIComponent(JSON.stringify({ fragment: uri.fragment, path: resourcePath }))}`);
```

Comment:
This is a lot of code for using an URI that goes like this: `file:///files/thisFile.md#L22`